bliss/generate.py

```diff
@@ -0,0 +1,98 @@
+import os
+import pickle
+from typing import Dict, List, TypedDict
+
+import torch
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+from torch.utils.data import IterableDataset
+from tqdm import tqdm
+
+from bliss.catalog import TileCatalog
+
+
+def generate(cfg: DictConfig):
+    simulator = instantiate(cfg.simulator)
+
+    file_data_capacity = cfg.cached_simulator.file_data_capacity
+    cached_data_path = cfg.cached_simulator.cached_data_path
+
+    # largest `batch_size` multiple <= `file_data_capacity`
+    file_data_size = (
+        file_data_capacity // simulator.image_prior.batch_size
+    ) * simulator.image_prior.batch_size
+
+    assert file_data_size >= simulator.image_prior.batch_size, "file_data_capacity too small"
+
+    # number of files needed to store >= `n_batches` * `batch_size` images
+    # in <= `file_data_size`-image files
+    n_files = -(
+        simulator.n_batches * simulator.image_prior.batch_size // -file_data_size
+    )  # ceil division
+
+    # stores details of the written image files - { filename: string, data }
+    data_files: List[Dict] = []
+
+    # assume overwriting existing files (if exist)
+    # use SimulatedDataset to generate data in minibatches iteratively,
+    # then concatenate before caching to disk via pickle
+    simulated_dataset = simulator.train_dataloader().dataset
+    assert isinstance(
+        simulated_dataset, IterableDataset
+    ), "simulated_dataset must be IterableDataset"
+
+    # create cached_data_path if it doesn't exist
+    if not os.path.exists(cached_data_path):
+        os.makedirs(cached_data_path)
+
+    if len(data_files) != n_files:
```

Comment:
When is this false? Only if `n_files == 0`? If so, it seems clearer to just use `assert n_files > 0`

Reply:
(also, then everything below wouldn't have to be indented as much)

Reply:
Done.